lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -41,13 +41,14 @@
   private final List<String> rackNames;
   private final String runtimeName;
   private final boolean relaxLocality;
+  private final SchedulingConstraint schedulingConstraint;
 
   EvaluatorRequest(final int number,
                    final int megaBytes,
                    final int cores,
                    final List<String> nodeNames,
                    final List<String> rackNames) {
-    this(number, megaBytes, cores, nodeNames, rackNames, "");
+    this(number, megaBytes, cores, nodeNames, rackNames, "", null, true);
```

Comment:
Do we need this constructors? They are only called from within the builder, right?